tools/codegen/gofilegen/datasource/data_source_file.go

```diff
@@ -42,14 +44,16 @@ func GeneratePluralGoCode(input *codespec.Resource) ([]byte, error) {
 
 	listOp := input.DataSources.Operations.List
 	pathParams := resource.GetPathParams(listOp.Path)
+	queryParams := getQueryParams(*input.DataSources.Schema.PluralDSAttributes)
 
 	tmplInputs := codetemplate.PluralDataSourceFileInputs{
 		PackageName:    input.PackageName,
-		DataSourceName: input.Name + "_list", // pluralize name
+		DataSourceName: pluralizeName(input.Name),
 		VersionHeader:  input.DataSources.Operations.VersionHeader,
 		ReadPath:       listOp.Path,
 		ReadMethod:     listOp.HTTPMethod,
 		PathParams:     pathParams,
+		QueryParams:    queryParams,
```

Comment:
could we add a unit test in `data_source_file.go`? Will help understand how resulting code looks, and we can also verify correctly handling for all different types.